testing/run_tests.py

```diff
@@ -605,25 +605,53 @@ def RunObjcTests(ios_variant='ios_debug_sim_unopt', test_filter=None):
   ios_out_dir = os.path.join(out_dir, ios_variant)
   EnsureIosTestsAreBuilt(ios_out_dir)
 
-  ios_unit_test_dir = os.path.join(
-      buildroot_dir, 'flutter', 'testing', 'ios', 'IosUnitTests'
-  )
+  new_simulator_name = 'IosUnitTestsSimulator'
 
-  # Avoid using xcpretty unless the following can be addressed:
-  # - Make sure all relevant failure output is printed on a failure.
-  # - Make sure that a failing exit code is set for CI.
-  # See https://github.com/flutter/flutter/issues/63742
-  command = [
-      'xcodebuild '
-      '-sdk iphonesimulator '
-      '-scheme IosUnitTests '
-      "-destination platform='iOS Simulator,name=iPhone 11' "
-      'test '
-      'FLUTTER_ENGINE=' + ios_variant
+  # Delete simulators with this name in case any were leaked
+  # from another test run.
+  DeleteSimulator(new_simulator_name)
+
+  create_simulator = [
+      'xcrun '
+      'simctl '
+      'create '
+      '%s com.apple.CoreSimulator.SimDeviceType.iPhone-11' % new_simulator_name
```

Comment:
If this line of code changes 634 has to change too.  Can the test command optionally take in the simulator name for the destination?  Otherwise at a minimum we can pull these out into variables that are next to each other so it's clear they are updated in tandem.

Reply:
Actually, I think we can get away with changing the 634 logic from `platform='iOS Simulator,name=iPhone 11'` to `name=IosUnitTestsSimulator`.  I forgot that was why I needed to delete the old simulators so there were never multiple with the same name, or the `xcodebuild` command would fail since there were multiple possible destinations.  Let me do that.

Reply:
(obligatory comment to make this a request)